src/libs/actions/Session/index.js

```diff
@@ -248,6 +257,53 @@ function signIn(password, validateCode, twoFactorAuthCode) {
     API.write('SigninUser', params, {optimisticData, successData, failureData});
 }
 
+function signInFromMagicLink(accountID, validateCode) {
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                ...CONST.DEFAULT_ACCOUNT_DATA,
+                isLoading: true,
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.CREDENTIALS,
+            value: {
+                validateCode,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    // this is temporary for now. Server should login with the accountID and validateCode
+    const params = {validateCode};
+    params.email = credentials.login;
+
+    API.write('SigninUser', params, {optimisticData, successData, failureData});
```

Comment:
NAB: `params` is used only once

```suggestion
    API.write('SigninUser', {
        validateCode,
        email: credentials.login,
    }, {optimisticData, successData, failureData});
```